ext/openssl/cipher/aead/aead.c

```diff
@@ -5,16 +5,15 @@
 VALUE dOSSL;
 VALUE eCipherError;
 
-#define GetCipherInit(obj, ctx) do {                    \
-        Data_Get_Struct((obj), EVP_CIPHER_CTX, (ctx));  \
-    } while (0)
-
-#define GetCipher(obj, ctx) do {                                        \
-        GetCipherInit((obj), (ctx));                                    \
-        if (!(ctx)) {                                                   \
-            ossl_raise(rb_eRuntimeError, "Cipher not inititalized!");   \
-        }                                                               \
-    } while (0)
+#define GetCipherInit(obj, ctx) do { \
+    TypedData_Get_Struct((obj), EVP_CIPHER_CTX, RTYPEDDATA_TYPE(obj), (ctx)); \
```

Comment:
yes, `ext/openssl` https://github.com/ruby/ruby/blob/trunk/ext/openssl/ossl_cipher.c#L12 , https://github.com/ruby/ruby/blob/trunk/ext/openssl/ossl_cipher.c#L88 && https://github.com/ruby/ruby/blob/trunk/ext/openssl/ossl_cipher.c#L936

This gem basically monkey patches a Data object, which strictly requires having to use the same typed data struct. Also upstream the parent of `Cipher` should be `rb_cData` and not `rb_cObject` as per the extension docs - https://github.com/ruby/ruby/blob/trunk/ext/openssl/ossl_cipher.c#L933 , https://github.com/ruby/ruby/blob/trunk/doc/extension.rdoc#c-struct-to-ruby-object :

```
It is recommended that klass derives from a special class called Data (rb_cData) but not from Object or other ordinal classes. If it doesn't, you have to call rb_undef_alloc_func(klass).
```